addon-test-support/@ember/test-helpers/setup-rendering-context.js

```diff
@@ -70,123 +165,58 @@ export default function setupRenderingContext(context) {
+      // `Ember._ContainerProxyMixin` and `Ember._RegistryProxyMixin` in this scenario we need to
+      // manually start the event dispatcher.
+      if (owner._emberTestHelpersMockOwner) {
+        let dispatcher = owner.lookup('event_dispatcher:main') || Ember.EventDispatcher.create();
+        dispatcher.setup({}, '#ember-testing');
+      }
 
-        toplevelView.setOutletState(outletState);
+      let OutletView = owner.factoryFor
+        ? owner.factoryFor('view:-outlet')
+        : owner._lookupFactory('view:-outlet');
+      let toplevelView = OutletView.create();
 
-        // using next here because the actual rendering does not happen until
-        // the renderer detects it is dirty (which happens on backburner's end
-        // hook), see the following implementation details:
-        //
-        // * [view:outlet](https://github.com/emberjs/ember.js/blob/f94a4b6aef5b41b96ef2e481f35e07608df01440/packages/ember-glimmer/lib/views/outlet.js#L129-L145) manually dirties its own tag upon `setOutletState`
-        // * [backburner's custom end hook](https://github.com/emberjs/ember.js/blob/f94a4b6aef5b41b96ef2e481f35e07608df01440/packages/ember-glimmer/lib/renderer.js#L145-L159) detects that the current revision of the root is no longer the latest, and triggers a new rendering transaction
-        return settled();
+      owner.register('-top-level-view:main', {
+        create() {
+          return toplevelView;
+        },
       });
-    };
 
-    if (global.jQuery) {
-      context.$ = function $(selector) {
-        // emulates Ember internal behavor of `this.$` in a component
-        // https://github.com/emberjs/ember.js/blob/v2.5.1/packages/ember-views/lib/views/states/has_element.js#L18
-        return selector ? global.jQuery(selector, context.element) : global.jQuery(context.element);
-      };
-    }
-
-    context.clearRender = function clearRender() {
-      return nextTickPromise().then(() => {
-        toplevelView.setOutletState({
-          render: {
-            owner,
-            into: undefined,
-            outlet: 'main',
-            name: 'application',
-            controller: context,
-            ViewClass: undefined,
-            template: OutletTemplate,
-          },
-          outlets: {},
-        });
+      // initially render a simple empty template
+      return render(hbs``).then(() => {
+        run(toplevelView, 'appendTo', getRootElement());
 
         return settled();
       });
-    };
-
-    return context;
-  });
+    })
+    .then(() => {
+      // ensure the element is based on the wrapping toplevel view
+      // Ember still wraps the main application template with a
+      // normal tagged view
+      //
+      // In older Ember versions (2.4) the element itself is not stable,
+      // and therefore we cannot update the `this.element` until after the
+      // rendering is completed
+      context.element = document.querySelector('#ember-testing > .ember-view');
```

Comment:
I assume this will break once the RFC from @cibernox is implemented?

Reply:
It probably will 🤔

Reply:
It just changes to use the rootElement directly (based on a feature detect):

```
context.element = getRootElement();
```

At which point the various interaction helpers (which were just changed to use `rootElement` instead of this) _and_ `this.element` will be back in sync...